modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py

```diff
@@ -1397,5 +1398,53 @@ def sort(df, cols, ascending, na_position, **kwargs):
     #    )
 
 
+class TestBadData:
+    bad_for_arrow = {
+        "a": ["a", [[1, 2], [3]], [3, 4]],
+        "b": ["b", [1, 2], [3, 4]],
+        "c": ["1", "2", 3],
+    }
+    bad_for_omnisci = {
+        "b": [[1, 2], [3, 4], [5, 6]],
+        "c": ["1", "2", "3"],
+    }
+    ok_data = {"d": np.arange(3), "e": np.arange(3), "f": np.arange(3)}
+
+    def _get_pyarrow_table(self, obj):
+        if not isinstance(obj, (pandas.DataFrame, pandas.Series)):
+            obj = pandas.DataFrame(obj)
+
+        return pyarrow.Table.from_pandas(obj)
+
+    @pytest.mark.parametrize("data", [bad_for_arrow, bad_for_omnisci])
+    def test_construct(self, data):
+        def applier(df, *args, **kwargs):
+            return repr(df)
+
+        run_and_compare(applier, data=data, force_lazy=False)
+
+    def test_from_arrow(self):
+        at = self._get_pyarrow_table(self.bad_for_omnisci)
+        md_df, pd_df = pd.utils.from_arrow(at), pandas.DataFrame(self.bad_for_omnisci)
```

Comment:
Two assignments would be easier to read.

Reply:
done